client/state/selectors/is-main-site-to-secondary-site-connection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import { getSiteOption, isJetpackSiteMainNetworkSite, isJetpackSiteSecondaryNetworkSite } from 'state/sites/selectors';
+import { withoutHttp } from 'lib/url';
+
+/**
+ * Returns true if site with id equal to mainSiteId is the main site of site with id equal to secondarySiteId and false otherwise.
+ * Returns null if with the information available in state the relationship is unknown.
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  mainSiteId      The ID of the main site
+ * @param  {Number}  secondarySiteId      The ID of the main site
+ * @return {?Boolean}            Whether site with id equal to mainSiteId is main site of site with id equal to secondarySiteId
+ */
+export default ( state, mainSiteId, secondarySiteId ) => {
```

Comment:
I like it that this selector will return `null` if we don't have any of those sites in the state. 

Would that make sense for the rest of the selectors that you're introducing here? What I mean specifically is:

* for `getNetworkSites`, return `null` if the site with the passed `id` is not found in the state.
* for `isConnectedSecondaryNetworkSite`, return `null` if any of the sites is not in the state.

This could be helpful when building UIs, so we know whether to show a placeholder or not.